subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomRelocationIntegrationTest.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.maven
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import spock.lang.Issue
+import spock.lang.Unroll
+
+class MavenPomRelocationIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    @Issue('https://github.com/gradle/gradle/issues/1789')
+    @Unroll
+    def "succeed when relocation exists"() {
+        given:
+        def original = publishPomWithRelocation('groupA', 'projectA', relocationGroupId, relocationArtifactId)
+        def newModule = mavenHttpRepo.module(newGroupId, newArtifactId, "1.0").publish()
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("${newArtifactId}-1.0.jar")
+
+        where:
+        relocationGroupId | relocationArtifactId | newGroupId  | newArtifactId
+        'groupB'          | 'projectB'           | 'groupB'    | 'projectB'
+        'newGroupA'       | ''                   | 'newGroupA' | 'projectA'
```

Comment:
I think this test case would become easier to read and understand if we'd use a common relocation group and artifact ID. Let's also a different naming for the artifact ID. At the moment you are using `project` but the published artifacts are not really projects in that sense. They are published modules. Coordinates that have no values should use null instead of an empty String. There's probably also another test case we are missing here if I am not mistaken.

Here're the test cases I am thinking of:


```
 'groupB' | 'artifactB' | 'groupB' | 'artifactB'
 'groupB' | null        | 'groupA' | 'artifactA'
 null     | 'artifactB' | 'groupA' | 'artifactB'
```